contracts/lib/TimeLockUpgradeV2.sol

```diff
@@ -0,0 +1,152 @@
+/*
+    Copyright 2018 Set Labs Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+pragma solidity 0.5.7;
+
+import { Ownable } from "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import { SafeMath } from "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+
+/**
+ * @title TimeLockUpgradeV2
+ * @author Set Protocol
+ *
+ * The TimeLockUpgradeV2 contract contains a modifier for handling minimum time period updates
+ *
+ * CHANGELOG:
+ * - Requires that the caller is the owner
+ * - New function to allow deletion of existing timelocks
+ */
+contract TimeLockUpgradeV2 is
+    Ownable
+{
+    using SafeMath for uint256;
+
+    /* ============ State Variables ============ */
+
+    // Timelock Upgrade Period in seconds
+    uint256 public timeLockPeriod;
+
+    // Mapping of upgradable units and initialized timelock
+    mapping(bytes32 => uint256) public timeLockedUpgrades;
+
+    /* ============ Events ============ */
+
+    event UpgradeRegistered(
+        bytes32 _upgradeHash,
+        uint256 _timestamp
+    );
+
+    event UpgradeUnregistered(
+        bytes32 _upgradeHash
+    );
+
+    /* ============ Modifiers ============ */
+
+    modifier timeLockUpgrade() {
```

Comment:
In the future when we make a V2 of something can the first commit just be copying the V1 into the V2 file so then we can have a record of changes after that? Makes much easier to see what the code changes were from V1 to V2.